lib/main.js

```diff
@@ -146,24 +156,57 @@ profile.registerSchema = function ( url, json ) {
   if (!_url2profile[url] || !json) {
     return;
   }
+
+  if ( url.indexOf('profile://') !== 0 ) {
+    console.error(`Failed to load profile ${url}: invalid protocol.`);
+    return;
+  }
+
+  let type = url.match(/^profile:\/\/([^\/]+)/)[1];
+
+  if (_type2parent[type]) {
+    console.log(`Warnning to register schema ${url}: The schema only takes effect on the root type`);
```

Comment:
console.warn(`Failed to register schema ${url}: only root profile can register schema`);